great_expectations/render/renderer/column_section_renderer.py

```diff
@@ -427,79 +437,23 @@ def _render_histogram(cls, evrs):
-            if chart_pixel_width < 200:
-                chart_pixel_width = 200
-            chart_container_col_width = round((len(weights) / 60.0) * 12)
-            if chart_container_col_width < 4:
-                chart_container_col_width = 4
-            elif chart_container_col_width > 8:
-                chart_container_col_width = 12
-            elif chart_container_col_width > 4:
-                chart_container_col_width = 8
-
-        mark_bar_args = {}
-        if len(weights) == 1:
-            mark_bar_args["size"] = 20
-
-        if kl_divergence_evr.result["details"]["observed_partition"].get("bins"):
-            bins = kl_divergence_evr.result["details"]["observed_partition"]["bins"]
-            bins_x1 = [round(value, 1) for value in bins[:-1]]
-            bins_x2 = [round(value, 1) for value in bins[1:]]
-
-            df = pd.DataFrame({
-                "bin_min": bins_x1,
-                "bin_max": bins_x2,
-                "fraction": weights,
-            })
-            df.fraction *= 100
-
-            bars = alt.Chart(df).mark_bar(**mark_bar_args).encode(
-                x='bin_min:O',
-                x2='bin_max:O',
-                y="fraction:Q",
-                tooltip=["bin_min", "bin_max", "fraction"]
-            ).properties(width=chart_pixel_width, height=400, autosize="fit")
-            chart = bars.to_json()
-        elif kl_divergence_evr.result["details"]["observed_partition"].get("values"):
-            values = kl_divergence_evr.result["details"]["observed_partition"]["values"]
-
-            df = pd.DataFrame({
-                "values": values,
-                "fraction": weights
-            })
-            df.fraction *= 100
 
-            bars = alt.Chart(df).mark_bar(**mark_bar_args).encode(
-                x='values:N',
-                y="fraction:Q",
-                tooltip=["values", "fraction"]
-            ).properties(width=chart_pixel_width, height=400, autosize="fit")
-            chart = bars.to_json()
-
-        return RenderedGraphContent(**{
-            "content_block_type": "graph",
-            "header":
-                {
-                    "template": "Histogram",
-                    "tooltip": {
-                        "content": "expect_column_kl_divergence_to_be_less_than"
-                    }
-                },
-            "graph": chart,
-            "styling": {
-                "classes": ["col-" + str(chart_container_col_width)],
-                "styles": {
-                    "margin-top": "20px",
+        header = RenderedStringTemplateContent(**{
+            "content_block_type": "string_template",
+            "string_template": {
+                "template": "Histogram",
+                "tooltip": {
+                    "content": "expect_column_kl_divergence_to_be_less_than"
                 }
             }
         })
 
+        return ExpectationStringRenderer._get_kl_divergence_chart(observed_partition_object, header)
```

Comment:
even though that's a classmethod now, I think we want to instantiate that (probably in init) and use the same object throughout as part of our new push to ensure configurability.